api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,15 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
+				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
+				&& this.getStartDate().equals(o.getStartDate())
+				&& ((this.getEndDate() != null && o.getEndDate() != null
+				&& this.getEndDate().equals(o.getEndDate()))
+				|| (this.getEndDate() == null && o.getEndDate() == null))) {
+			return 0;
+		} else {
+			return -1;
+		}
```

Comment:
> And always push stopped memberships to the end of the collections with the one with the latest start date coming dead last.

Do you want this in the compareTo() method? I was trying to find a way to do this: One method is to execute [Collections.rotate(List, -1) ](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#rotate(java.util.List,%20int)). Is there any other method?